06-building-a-program-start-to-finish/writing-tests/output/invertedIndex-spec.js

```diff
@@ -0,0 +1,205 @@
+const values = require('object.values');  //  shim Object.values
+//  book with valid contents
+const book = require('./allBooks/books.json');
+//  book with invalid content
+const invalidFile = require('./allBooks/invalid.json');
+//  book with valid content
+const secondBook = require('./allBooks/newBook.json');
+//  empty book
+const empty = require('./allBooks/empty.json');
+// an array
+const invalid = require('./allBooks/in-valid.json');
+// empty array
+const emptyFile = require('./allBooks/emptyArray.json');
+// an invalid JSON file
+const notValid = require('./allBooks/notArrayOfArray.json');
+
+describe('invertedIndex Index', () => {
+   beforeAll(() => {
+    this.invertedIndex = new InvertedIndex();
+  });
+  
+  describe('Constructor', () => {
+    it('can create instances of inverted index class', () => {
+      expect(typeof this.invertedIndex)
+        .toEqual('object');
+      expect(this.invertedIndex instanceof InvertedIndex)
+        .toBe(true);
+    });
+  });
+
+  const invertedIndex = new InvertedIndex();
+  invertedIndex.createIndex('book.json', book);
+  invertedIndex.createIndex('secondBook.json', secondBook);
```

Comment:
Why are you creating the indices here. I'd rather you did that in the suite that tests for this.